### ISSUE TITLE
How to delete Annotations

### ISSUE THREAD

**Issue by nicHoch** [theCrag/website#2960]
**What happened?**
It is not obvious possible to delete Annotations. 
Some people bypath that with clearing name and description. 
Now the annotation is not visible on the website anymore But i am pretty sure I have seen such orphans in the index (API)

**What you expected:**

a proper and direct way to delete annotations.

**Comment by brendanheywood** [comment]
Cross linking to https://github.com/theCrag/website/issues/1011

We need a proper soft delete across the board for all object types.

**Comment by scd** [comment]
A couple of comments. 

Currently the way to delete is to remove the title and description and the system will interpret this as a delete. Actually, in this case it is a soft delete.

There are quite a few people who get confused with this so we should make it more explicit.

If there are any orphans then this might be a bug, so @nicHoch if you come across anything suspicious (ie annotations with empty title and description) then please log.

@brendanheywood I think #1011 should mostly be done. I will have a look at that in more detail.

**Comment by scd** [comment]
+1 user email not knowing how to delete an annotation

**Comment by scd** [comment]
@brendanheywood I was wrong, we have not completed #1011, although a lot of the ground work is done.

**Comment by brendanheywood** [comment]
+1 from support email

**Comment by cams-884** [comment]
I was having trouble reordering some annotations that got out of order after a reparenting on [this page](https://www.thecrag.com/climbing/australia/tasmania/area/1951294164). So I tried deleting them and re-adding them instead.

This worked for 3 of 4. The one that didn't work (Terrace Wall) is complaining about the name already existing when I try and add it again.

![image](https://user-images.githubusercontent.com/4734293/44317586-72dcc500-a475-11e8-9361-507b4c4cdf9f.png)

After I deleted the annotation I noticed that there was still a (now dead) link to it in the description of [one of the routes](https://www.thecrag.com/climbing/australia/tasmania/route/1880192301). Possibly the link prevented the annotation from deleting? I tried editing the route and removing the link. This didn't help.

I have added a placeholder annotation "Terrace Wall1" in the meantime.

Browser: Chrome Version 68.0.3440.106

**Comment by brendanheywood** [comment]
Yeah that definitely sounds like the soft delete isn't working properly, or at least the duplicate name detection is ignoring the deleted status.

The dud link to terrace wall (to here: https://www.thecrag.com/climbing/australia/tasmania/area/1944320949) I think is actually a different bug again, it's linking to the old area node which was merged??

@scd this one is for you, there is 3 different overlapping bugs. I think the quickest fix would be to find the duplicate annotation and just hack it's name to something else so that the current annotation can be renamed properly, and then we can fix the deeper bugs later.

**Comment by Luen** [comment]
+1

**Comment by nicHoch** [comment]
+1 mail

**Comment by rouletout** [comment]
Is this ok now?

**Comment by matozoid** [comment]
Having added and deleted quite a few annotations over the last months, I would love to have a "delete" option in the dropdown menu on the right of it.

**Comment by FreddieChopin** [comment]
> My suspect process is that bulk edit

Same here. When making an annotation empty via bulk edit, the annotation is not deleted. It is not visible, but when you do bulk edit again, it is still there, with the old name visible in the header and all fields empty. Adding any text to name field, saving, then editing just the annotation, clearing the name and saving really removes it.

**Comment by killakalle** [comment]
+1

Came across this problem after deciding that annotations within a sector were better suited as their own sub-sectors. After not being able to delete the annotations and googling "thecrag delete annotation" I found this issue.

PS: Following the tipps here, I edited the annotations and deleted the texts. The annotations were then removed and also not visible during Bulk Edit of the former parent node.

**Comment by rouletout** [comment]
Is this still an issue? I delete annotations every now and then and never have an issue....

**Comment by quaestor** [comment]
Yes it's an issue, in combination with #3753.
